internal/bzlmod/go_deps.bzl

```diff
@@ -220,8 +214,8 @@ def _go_deps_impl(module_ctx):
                 outdated_direct_dep_printer = fail
 
         _process_overrides(module_ctx, module, "gazelle_override", gazelle_overrides, _process_gazelle_override)
-        _process_overrides(module_ctx, module, "module_override", module_overrides, _process_module_override, archive_overrides)
-        _process_overrides(module_ctx, module, "archive_override", archive_overrides, _process_archive_override, module_overrides)
+        _process_overrides(module_ctx, module, "module_override", module_overrides, _process_module_override)
+        _process_overrides(module_ctx, module, "archive_override", archive_overrides, _process_archive_override)
```

Comment:
What's a use case for mixing `module_override` and `archive_override`? That logically seems wrong to me, maybe we should rather duplicate some of the attrs?

Reply:
I have a Go module that's downloaded from a specific `url` (not through GoProxy) and then I need to add some gazelle directives and apply some patches to make it build correctly with Bazel.

Reply:
I see why you need to combine `archive_override` with `gazelle_override` in that case, but why the need for `module_override`? I think that we should add any of its capabilities you need here to `archive_override` instead: `archive_override` and `module_override` both allow specifying a different source for the Go modules, having both could result in breakages or confusing behavior.

Reply:
I see what you mean. I updated this to better follow the design of the normal `archive_override` from Bzlmod